kratos/tests/test_serializer.py

```diff
@@ -0,0 +1,74 @@
+    from KratosMultiphysics.FluidDynamicsApplication.fluid_dynamics_analysis import FluidDynamicsAnalysis
+
+
+def GetFilePath(fileName):
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)), fileName)
+
+class TestSerializer(KratosUnittest.TestCase):
+    def _prepare_test(self):
+        # Define a model and load the parameters
+        self.pre_serialized_model = KratosMultiphysics.Model()
+        with open("auxiliar_files_for_python_unnitest/parameters_files/test_serializer.json",'r') as parameter_file:
+            parameters = KratosMultiphysics.Parameters(parameter_file.read())
+
+        # First the model is initialized
+        self.pre_serialized_simulation = FluidDynamicsAnalysis(self.pre_serialized_model, parameters)
+        self.pre_serialized_simulation.Initialize()
+
+        # Before serializing the model, main model part is set to RESTARTED
+        self.main_model_part_name = parameters["solver_settings"]["model_part_name"].GetString()
+        self.pre_serialized_model.GetModelPart(self.main_model_part_name).ProcessInfo.SetValue(KratosMultiphysics.IS_RESTARTED,True)
+        serialized_model = KratosMultiphysics.StreamSerializer()
+        serialized_model.Save("ModelSerialization",self.pre_serialized_model)
+
+        with open("auxiliar_files_for_python_unnitest/parameters_files/test_serializer.json",'r') as parameter_file:
+            self.project_parameters = KratosMultiphysics.Parameters(parameter_file.read())
+        # Parameters are read again and input type set to use_input_model_part since the serialized model already has the mdpa loaded
+        self.project_parameters["solver_settings"]["model_import_settings"]["input_type"].SetString("use_input_model_part")
+
+        # Deserialize and store the new model
+        self.current_model = KratosMultiphysics.Model()
+        serialized_model.Load("ModelSerialization",self.current_model)
+        del(serialized_model)
+
+    def _check_results(self):
+        pre_serialized_model_part = self.pre_serialized_model.GetModelPart(self.main_model_part_name).GetSubModelPart("NoSlip2D_structure")
+        pre_serialized_pressure_results = []
+        for node in pre_serialized_model_part.Nodes:
+            pressure = node.GetSolutionStepValue(KratosMultiphysics.PRESSURE)
+            pre_serialized_pressure_results.append(pressure)
+        serialized_model_part = self.current_model.GetModelPart(self.main_model_part_name).GetSubModelPart("NoSlip2D_structure")
+        serialized_pressure_results = []
+        for node in serialized_model_part.Nodes:
+            pressure = node.GetSolutionStepValue(KratosMultiphysics.PRESSURE)
+            serialized_pressure_results.append(pressure)
+        # Comparing results before and after serializing
+        for pre_serialized_result, serialized_result in zip(pre_serialized_pressure_results,serialized_pressure_results):
+            self.assertAlmostEqual(pre_serialized_result, serialized_result)
```

Comment:
+1 for using `zip`

why do you only compare the pressure?
I guess it is enough, just asking

Reply:
This geometry is a 2d box under the action of a fluid, so I chose to check the pressure on the nodes of the structure at first so it didn't add too much overhead.

But checking it now, it doesn't make too much of a difference so I will check both pressure and velocity for the whole domain